### PR TITLE
[NBS 1.0] Deprecate `techdocs-backend` create router

### PR DESCRIPTION
**Pull request by camilaibs**
Signed-off-by: Camila Belo <camilaibs@gmail.com>

## Hey, I just made a Pull Request!

Deprecating `createRouter` in preparation for ending support for the legacy backend system.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs-backend/package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
   },
   "dependencies": {
     "@backstage/backend-common": "workspace:^",
+    "@backstage/backend-defaults": "workspace:^",
```
Comment (Rugvip) [inline plugins/techdocs-backend/package.json:62]: Nop this dependency is not alright 😅, prolly fixed by switching to the cache service for `createRouter` instead

File: plugins/techdocs-backend/src/service/router.test.ts [hunk 2]
```diff
@@ -76,7 +74,11 @@ const getMockHttpResponseFor = (content: string): Buffer => {
 const createApp = async (options: RouterOptions) => {
   const app = express();
   app.use(await createRouter(options));
-  app.use(errorHandler());
+  const errorHandler = MiddlewareFactory.create({
+    config: mockServices.rootConfig(),
+    logger: mockServices.rootLogger(),
+  }).error();
```
Comment (Rugvip) [inline plugins/techdocs-backend/src/service/router.test.ts:80]: Let's use the new util here instead

File: plugins/techdocs-backend/package.json (hunk 1 above)
Comment (camilaibs) [reply plugins/techdocs-backend/package.json]: Change reverted in the create router options.

File: plugins/techdocs-backend/src/service/router.test.ts (hunk 2 above)
Comment (camilaibs) [reply plugins/techdocs-backend/src/service/router.test.ts]: Done 🙏🏻

**Review by Rugvip — APPROVED** [review]
Looks good 👍

**Review by byan1197 — APPROVED** [review]

**Pull request merged**